app/models/order_detail.rb

```diff
@@ -752,7 +762,9 @@ def cancel_reservation(canceled_by, canceled_reason: nil, order_status: OrderSta
         change_status! order_status
       end
     else
+      return false if state.to_sym == :unrecoverable
```

Comment:
Why is this here?

Reply:
I don't know if we'll have this case, but it made sense to prevent a reservation from being canceled if the work had already been done